src/main/java/org/truffleruby/core/CoreLibrary.java

```diff
@@ -210,7 +212,8 @@
     @CompilationFinal private DynamicObject eagainWaitReadable;
     @CompilationFinal private DynamicObject eagainWaitWritable;
 
-    private final Map<Errno, DynamicObject> errnoClasses = new HashMap<>();
+    private final Map<String, DynamicObject> errnoClasses = new HashMap<>();
+    private final Map<Integer, String> errnoValueToNames = new HashMap<>();
```

Comment:
I'm fine with a map here, but are the errno values sequential enough where a `String[]` could be used instead?

Reply:
How can I know what is the maximum errno value?
I would need an extra iteration, which is not so cheap with `getSection()`.
Or maybe I should use an ArrayList then and let it resize as needed?

Reply:
This is fine then. I didn't know if POSIX guarantees they're < 1024 or what have you.

Reply:
I tried an ArrayList but the interface doesn't seem to allow to use it nicely like a Map.
One needs to call `add()` `n-1` times to increment the size so it lets you add an element at position `n` without throwing an exception.
This complicates the code significantly and might be fairly inefficient.
So I'll keep the Map for now.
We could have our own "Ruby-like Array which grows as needed and allows set(index, value) with any index>=0" if we care enough.